frontend/app_flowy/packages/flowy_editor/example/lib/plugin/selected_text_node_widget.dart

```diff
@@ -0,0 +1,261 @@
+import 'package:flowy_editor/flowy_editor.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:url_launcher/url_launcher_string.dart';
+
+class SelectedTextNodeBuilder extends NodeWidgetBuilder {
+  SelectedTextNodeBuilder.create({
+    required super.node,
+    required super.editorState,
+    required super.key,
+  }) : super.create() {
+    nodeValidator = ((node) {
+      return node.type == 'text';
+    });
+  }
+
+  @override
+  Widget build(BuildContext buildContext) {
+    return _SelectedTextNodeWidget(
+      key: key,
+      node: node,
+      editorState: editorState,
+    );
+  }
+}
+
+class _SelectedTextNodeWidget extends StatefulWidget {
+  final Node node;
+  final EditorState editorState;
+
+  const _SelectedTextNodeWidget({
+    Key? key,
+    required this.node,
+    required this.editorState,
+  }) : super(key: key);
+
+  @override
+  State<_SelectedTextNodeWidget> createState() =>
+      _SelectedTextNodeWidgetState();
+}
+
+class _SelectedTextNodeWidgetState extends State<_SelectedTextNodeWidget>
+    with Selectable {
+  TextNode get node => widget.node as TextNode;
+  EditorState get editorState => widget.editorState;
+
+  final _textKey = GlobalKey();
+  TextSelection? _textSelection;
+
+  RenderParagraph get _renderParagraph =>
+      _textKey.currentContext?.findRenderObject() as RenderParagraph;
+
+  @override
+  List<Rect> getSelectionRectsInSelection(Offset start, Offset end) {
+    var textSelection =
+        TextSelection(baseOffset: 0, extentOffset: node.toRawString().length);
+    // Returns select all if the start or end exceeds the size of the box
+    // TODO: don't need to compute everytime.
+    var rects = _computeSelectionRects(textSelection);
+    _textSelection = textSelection;
+
+    if (end.dy > start.dy) {
+      // downward
+      if (end.dy >= rects.last.bottom) {
+        return rects;
+      }
+    } else {
+      // upward
+      if (end.dy <= rects.first.top) {
+        return rects;
+      }
+    }
+
+    final selectionBaseOffset = _getTextPositionAtOffset(start).offset;
+    final selectionExtentOffset = _getTextPositionAtOffset(end).offset;
+    textSelection = TextSelection(
+      baseOffset: selectionBaseOffset,
+      extentOffset: selectionExtentOffset,
+    );
+    _textSelection = textSelection;
+    return _computeSelectionRects(textSelection);
+  }
+
+  @override
+  Rect getCursorRect(Offset start) {
+    final selectionBaseOffset = _getTextPositionAtOffset(start).offset;
+    final textSelection = TextSelection.collapsed(offset: selectionBaseOffset);
+    _textSelection = textSelection;
+    return _computeCursorRect(textSelection.baseOffset);
+  }
+
+  @override
+  TextSelection? getTextSelection() {
+    return _textSelection;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    print('text rebuild $this');
```

Comment:
`debugPrint`

Reply:
OK

Reply:
> `debugPrint`

How to disable the debug print if the user doesn't want to print the log. Is there something like env print?